### DISCUSSION TITLE
openssl/macros.h(139): fatal error C1189: #error:  "The requested API level higher than the configured API compatibility level"

### DISCUSSION THREAD

**Discussion by lihao20110822** [tianocore/edk2#6068] [Q&A]
I tried to build edk2,but I met an error:
e:\uefiworkstation\edk2\CryptoPkg\Library\OpensslLib\openssl\include\openssl/macros.h(139): fatal error C1189: #error:  "The requested API level higher than the configured API compatibility level"

I uses the Visual Studio 2017,can anyone help me?
![屏幕截图 2024-08-11 104530](https://github.com/user-attachments/assets/5933636d-0c5d-4843-9810-6eb0e5211ed0)

**Comment by xin-hai-WU** [comment]
I encountered the same problem as you.
![image](https://github.com/user-attachments/assets/9f085370-cefe-4bcb-9273-8baaacc13460)
